.github/workflows/deploy-ghpages-latest-stable.yml

```diff
@@ -0,0 +1,151 @@
+name: Deploy Sphinx documentation
+
+on:
+
+  workflow_call:
+    inputs:
+        python-version:
+          required: true
+          type: string
+        package-manager:
+          required: true 
+          type: string
+        dependency-path:
+          required: true 
+          type: string
+        trigger-label:
+          required: false
+          type: string
+          default: none
+        path-doc:
+          required: false
+          type: string 
+          default: ./doc
+
+        
+          
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow one concurrent deployment
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
+
+jobs:
+  check-label:
+    runs-on: ubuntu-latest
+    outputs:
+      label: ${{ steps.valid_label.outputs.value}}
+    steps:
+      - name: checks for the label 
+        id: valid_label
+        run: |
+          if [[ "${{ inputs.trigger-label }}" == "latest" ]] || [[ "${{ inputs.trigger-label  }}" == "stable" ]];
+            then
+                echo "value=true" >> $GITHUB_OUTPUT
+            else
+                echo "value=false" >> $GITHUB_OUTPUT 
+          fi
+            
+  
+  build:
+    
+    runs-on: ubuntu-latest
+    needs: check-label
+    if: needs.check-label.outputs.label == 'true'
```

Comment:
You do not need to check the output: if the job required in `needs` is failing, this will not keep going.
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds